sway-ir/tests/serialize/intrinsic_addr_of.ir

```diff
@@ -0,0 +1,26 @@
+// regex: VAR=v\d+
+
+script {
+  // check: main() -> u64
+  fn main() -> u64, !1 {
+        local ptr { u64, ( { b256 } | { b256 } ) } sender
+
+        entry:
+        // check: = get_ptr
+        v0 = get_ptr ptr { u64, ( { b256 } | { b256 } ) } sender, ptr { u64, ( { b256 } | { b256 } ) }, 0, !2
+        v1 = const { u64, ( { b256 } | { b256 } ) } { u64 0, { b256 } { b256 0x0100000000000000000000000000000000000000000000000000000000000010 } }, !4
+        store v1, ptr v0, !2
+        // check: $(arg0=$VAR) = get_ptr
+        v2 = get_ptr ptr { u64, ( { b256 } | { b256 } ) } sender, ptr { u64, ( { b256 } | { b256 } ) }, 0, !5
+        // check: addr_of $arg0
+        v3 = addr_of v2
+        ret u64 v3
+    }
+}
+
+!0 = "/home/vaivaswatha/c/sway/test/src/e2e_vm_tests/test_programs/should_pass/language/addrof_intrinsic/src/main.sw"
```

Comment:
You're leaking some personal directory stuff here.  Probably doesn't matter but it's a little weird to keep in the repo.

Since you're not actually testing metadata I'd just strip all the metadata out.  You could strip a lot of it out -- you're just testing that `addr_of` is parsed and re-printed correctly, so it only really needs a `get_ptr` and the `addr_of` instructions, none of the rest...

Reply:
> You're leaking some personal directory stuff here. Probably doesn't matter but it's a little weird to keep in the repo.
> 
> Since you're not actually testing metadata I'd just strip all the metadata out. You could strip a lot of it out -- you're just testing that `addr_of` is parsed and re-printed correctly, so it only really needs a `get_ptr` and the `addr_of` instructions, none of the rest...

Ah I overlooked this. Fixing it now.